### PR TITLE
feat: change the address of the commission contract

### PR DESCRIPTION
**Pull request by CTAPCKPIM**

### CODE REVIEW

**Comment by CTAPCKPIM** [comment]
- I will correct the mistake now

File: contracts/StreamManager.sol [hunk 1]
```diff
@@ -42,6 +42,11 @@ contract StreamManager is IStreamManager, ReentrancyGuard {
      * @param _payer new address  
      */
     event PayerChanging(address _payer);
+    /**
+     * @dev Changing address of fee
+     * @param _feeAddress new address of the fee
+     */
+    event AddressFeeChanging(address _feeAddress);
```
Comment (markholdex) [inline contracts/StreamManager.sol:49]: @CTAPCKPIM please use past tense for the event names. And update accordingly. Thanks!

File: contracts/StreamManager.sol [hunk 2]
```diff
@@ -83,6 +90,7 @@ contract StreamManager is IStreamManager, ReentrancyGuard {
     constructor(address _payer) {
         payer = _payer;
         admin = msg.sender;
+        feeAddress = msg.sender;
```
Comment (markholdex) [inline contracts/StreamManager.sol:93]: @CTAPCKPIM we already had the admin address as the fees address. Why did you decide to change that? What is the purpose of the admin address now?

**Review by markholdex — CHANGES_REQUESTED** [review]
Please review comments

File: contracts/StreamManager.sol (hunk 2 above)
Comment (CTAPCKPIM) [reply contracts/StreamManager.sol:93]: @markholdex 
I misunderstood the role of the administrator address, in which case I will fix it

**Review by zolotokrylin — CHANGES_REQUESTED** [review]
tests please

File: contracts/StreamManager.sol [hunk 3]
```diff
@@ -280,4 +285,13 @@ contract StreamManager is IStreamManager, ReentrancyGuard {
         payer = _payer;
         emit PayerChanging(_payer);
     }
+
+    ///@dev changing address of the fee
+    function chaingeAddressFee(address _feeAddress) public onlyAdmin {
+        if (_feeAddress == address(0)) revert InvalidAddress();
+        if (_feeAddress == admin) revert InvalidAddress();
```
Comment (zolotokrylin) [inline contracts/StreamManager.sol:292]:
```suggestion
```

It is ok to be admin address

**Comment by CTAPCKPIM** [comment]
@zolotokrylin / @markholdex 

Done, check pls

File: test/StreamManager.test.js [hunk 4]
```diff
@@ -382,4 +382,64 @@ describe("StreamManager:", function () {
     // Recall should return an error
     expect(this.streamManager.connect(this.payee4).claim()).to.be.revertedWith('ReentrancyGuardReentrantCall')
   })
+
+  // Tests for `changePayer();`
+  // Changing the address payer
```
Comment (zolotokrylin) [inline test/StreamManager.test.js:387]:
```suggestion
  // Changing the address the payer
```

File: test/StreamManager.test.js [hunk 5]
```diff
@@ -382,4 +382,64 @@ describe("StreamManager:", function () {
...
+  it.only('changePayer: address of the payer is changing', async () => {
+    await expect(
+      this.streamManager.connect(this.admin).changePayer(this.payee1.address)
+    ).to.emit(this.streamManager, "PayerChanging")
+    .withArgs(this.payee1.address);
+  })
+
+  // Expecting revert with `NotAdmin`
+  it.only('changePayer: only the admin can call the function', async () => {
```
Comment (zolotokrylin) [inline test/StreamManager.test.js:396]:
```suggestion
  it.only('changePayer: only the admin can call this function', async () => {
```

File: test/StreamManager.test.js [hunk 6]
```diff
@@ -382,4 +382,64 @@ describe("StreamManager:", function () {
...
+    await expect(
+      this.streamManager.connect(this.payee4).changePayer(this.payee1.address)
+      ).to.be.revertedWith('NotAdmin')
+  })
+
+  // Expecting revert with `InvalidAddress`
+  it.only('changePayer: not can setting address(0) how address of the payer', async () => {
```
Comment (zolotokrylin) [inline test/StreamManager.test.js:403]:
```suggestion
  it.only('changePayer: can not set the address to address(0)', async () => {
```

File: test/StreamManager.test.js [hunk 7]
```diff
@@ -382,4 +382,64 @@ describe("StreamManager:", function () {
...
+    await expect(
+      this.streamManager.connect(this.admin).changePayer(this.zero)
+      ).to.be.revertedWith('InvalidAddress')
+  })
+
+  // Expecting revert with `InvalidAddress`
+  it.only('changePayer: not can setting same address how the address what to was setting', async () => {
```
Comment (zolotokrylin) [inline test/StreamManager.test.js:410]: Can you please rephrase this one? I can't understand what is the test about

File: test/StreamManager.test.js [hunk 8]
```diff
@@ -382,4 +382,64 @@ describe("StreamManager:", function () {
...
+    await expect(
+      this.streamManager.connect(this.admin).changePayer(this.payee1.address)
+      ).to.be.revertedWith('InvalidAddress')
+  })
+
+  // Tests for `chaingeAddressFee();`
+  // Changing the address payer
+  it.only('changePayer: address of the admin is changing', async () => {
```
Comment (zolotokrylin) [inline test/StreamManager.test.js:418]: why changePayer? should it be change commisionAddress?

File: test/StreamManager.test.js [hunk 9]
```diff
@@ -382,4 +382,64 @@ describe("StreamManager:", function () {
...
+    await expect(
+      this.streamManager.connect(this.admin).chaingeAddressFee(this.payee1.address)
```
Comment (zolotokrylin) [inline test/StreamManager.test.js:420]:
```suggestion
      this.streamManager.connect(this.admin).changeAddressFee(this.payee1.address)
```

Comment (zolotokrylin) [reply test/StreamManager.test.js:420]: and rename it please into changeComissionAddr

File: test/StreamManager.test.js [hunk 10]
```diff
@@ -382,4 +382,64 @@ describe("StreamManager:", function () {
...
+
+  // Expecting revert with `NotAdmin`
+  it.only('chaingeAddressFee: only the admin can call the function', async () => {
+    await expect(
+      this.streamManager.connect(this.payee4).chaingeAddressFee(this.payee1.address)
+      ).to.be.revertedWith('NotAdmin')
+  })
+
+  // Expecting revert with `InvalidAddress`
+  it.only('chaingeAddressFee: not can setting address(0) how address of the admin', async () => {
```
Comment (zolotokrylin) [inline test/StreamManager.test.js:433]:
```suggestion
  it.only('chaingeAddressFee: can not set address to zero', async () => {
```

File: contracts/StreamManager.sol [hunk 11]
```diff
@@ -42,6 +42,11 @@ contract StreamManager is IStreamManager, ReentrancyGuard {
      * @param _payer new address  
      */
     event PayerChanging(address _payer);
```
Comment (markholdex) [inline contracts/StreamManager.sol:44]:
```suggestion
    event PayerAddressChanged(address _payer);
```

- Also update in other places

File: contracts/StreamManager.sol [hunk 12]
```diff
@@ -42,6 +42,11 @@ contract StreamManager is IStreamManager, ReentrancyGuard {
...
+    /**
+     * @dev Changing address of fee
+     * @param _feeAddress new address of the fee
+     */
+    event AddressFeeChanged(address _feeAddress);
```
Comment (markholdex) [inline contracts/StreamManager.sol:49]:
```suggestion
    event ComissionAddressChanged(address _feeAddress); 
```
- update in other places too

File: contracts/StreamManager.sol [hunk 13]
```diff
@@ -280,4 +285,12 @@ contract StreamManager is IStreamManager, ReentrancyGuard {
         payer = _payer;
         emit PayerChanging(_payer);
     }
+
+    ///@dev changing address of the fee
+    function chaingeAddressFee(address _feeAddress) public onlyAdmin {
```
Comment (markholdex) [inline contracts/StreamManager.sol:290]:
```suggestion
    function changeComissionAddress(address _feeAddress) public onlyAdmin {
```

- update in other places where necessary. Also fixed a typo. It's not `chainge` it's `change`
- new name suggested by Vadim

File: contracts/StreamManager.sol [hunk 14]
```diff
@@ -280,4 +285,12 @@ contract StreamManager is IStreamManager, ReentrancyGuard {
...
+        if (_feeAddress == address(0)) revert InvalidAddress();
```
Comment (markholdex) [inline contracts/StreamManager.sol:291]: Want to add a verification for existing address?

File: test/StreamManager.test.js (hunk 7 above)
Comment (markholdex) [reply test/StreamManager.test.js:410]: Vadim is right here. @CTAPCKPIM This PR should only be related to Comission address and not Payer address.

**Review by markholdex — CHANGES_REQUESTED** [review]
@CTAPCKPIM please have a look at these requested changes. Let me know if anything unclear.

**Comment by CTAPCKPIM** [comment]
- I'll fix everything now

**Comment by CTAPCKPIM** [comment]
@zolotokrylin / @markholdex 

I fixed all, check pls

**Review by markholdex — APPROVED** [review]

**Pull request merged**